Pr/1

TylerBarnes opened the pull request:
This PR builds off of your work @jasonbahl and flips things around so Gatsby sends required version ranges to WPGatsby instead of sending it's own version.

Closes/merges https://github.com/TylerBarnes/gatsby/pull/1

jasonbahl committed:
Add Compatibility API to WPGatsby

- This updates the Schema to expose a new `RootQuery.wpGatsbyCompatibility` entry to query compatibility with a given version of gatsby-source-wordpress
- When querying, a version is required input, and that version is checked first against a remote compatibility file, then falls back to checking a local file. If the installed version of WPGraphQL or WPGatsby is outside the min/max specified versions, it will be marked as `isCompatible: false` and message(s) will be returned with further information

Example:

```
{
  isWpGatsby
  wpGatsbyCompatibility(version: "0.0.27") {
    messages
    isCompatible
  }
}
```

TylerBarnes committed:
flip things around on their head. Gatsby asks if it's requirements are met rather than the other way around

TylerBarnes committed:
add compatibility api to the source plugin

TylerBarnes committed:
Merge branch 'feat/source-wordpress-v4' into pr/1

jasonbahl commented on packages/wp-gatsby/src/Schema/SiteMeta.php:
```diff
@@ -14,16 +16,136 @@ function __construct()
+          $wpgatsby_version_range = $args['wpGatsbyVersionRange'] ?? null;
+
+
+			    if ( ! $wpgql_version_range ) {
```
spacing looks way off in here

jasonbahl commented on packages/wp-gatsby/wp-gatsby.php:
```diff
@@ -67,7 +68,7 @@ public function __clone()
 
         // Cloning instances of the class is forbidden.
-        _doing_it_wrong(__FUNCTION__, esc_html__('The WP_GATSBY_ class should not be cloned.', 'wp-gatsby'), '0.0.1');
+        _doing_it_wrong(__FUNCTION__, esc_html__('The WPGATSBY_ class should not be cloned.', 'wp-gatsby'), '0.0.1');
```
extra underscore here

jasonbahl commented on packages/wp-gatsby/src/Schema/SiteMeta.php:
```diff
@@ -14,16 +16,136 @@ function __construct()
+			     * Return the payload
+			     */
+			    return [
+			    	'satisfies' => [
```
spacing is weird

jasonbahl reviewed (changes requested):
Couple small changes. I'll go ahead and pull this down, make the changes and push

jasonbahl committed:
- Updated formatting / spacing
- Set WPGATSBY_VERSION constant to 0.2.0
- Removed unnecessary underscore from mention of WPGATSBY class (was WPGATSBY_)
